test/runtests.jl

```diff
@@ -273,4 +273,71 @@ end
     end
 end
 
+
+@testset "Callable kernel" begin
+    @testset "Trivial data" begin
+        X = [-2 -1 -1 1 1 2;
+             -1 -1 -2 1 2 1]
+        y = [1, 1, 1, 2, 2, 2]
+
+        kernel(x1, x2) = x1' * x2
+
+        model = svmtrain(X, y, kernel=kernel)
+
+        @test model.rho ≈ [0]
+        @test model.coefs ≈ [0.25; -0.25]
+        @test model.SVs.indices == [2, 4]
+
```

Comment:
I suggest adding a comment explaining source of the reference result. (I'm guessing they come from using the built-in linear kernel, yes?)

Reply:
Nice to have these other tests, by the way.

Reply:
This test (and the `"Iris data"` test) are adapted from tests for the precomputed kernel. Maybe @barucden can pitch in on where the coefficients come from.

Reply:
I am pretty sure I took it from https://github.com/JuliaML/LIBSVM.jl/pull/88/files